### DISCUSSION TITLE
Allow ".sortspec.md" name use? And how to hide sortspec note

### DISCUSSION THREAD

**Discussion by J3ster1337** [SebastianMC/obsidian-custom-sort#92] [Q&A] [answered]
Can you please make plugin search for ".sortspec.md" file alongside classic "sortspec.md"?
It will be better 'cause it will allow hiding file in Obsidian file view. Currently making this throws "not found custom sort file" error on reenabling sorting.
And thank you for creating this! Total must-have

**Comment by SebastianMC** [comment]
Hi @J3ster1337 

unfortunately, at technical level the hidden files (e.g. name starting with a dot) are not included by Obsidian in the set of 'vault files'. And the plugin scans the 'vault files'.

At the same time, you can get similar result by explicitly hiding the `sortspec.md` file inside the `sorting-spec:` itself:

```yaml
---
sorting-spec: |
  .... here go some sorting specifications, and below you can:
 
  /--hide: sortspec.md
---
```

Does it satisfy your needs? The additional advantage is the simple edit access to files hidden in such a way: turn off the custom sort (e.g. with ribbon toggle), the file will reappear, you can edit it directly in obsidian, then apply the updated sorting with ribbon toggle.
I know that this could be cumbersome if you have many sortspec.md files in your vault folders tree - the syntax of `/--hide:` requires providing the file name, no wildcards. Yet implementation of this can be simply extended to hide all files with matching name in the entire vault folder tree.

Please let me know if it helps

I'm aware that there are more advanced scenarios or cases where none of the above solutions are satisfactory.
Long term, I noticed in Obsidian API exposed to the plugins that a more direct access to underlying file system is possible, so perhaps scanning the `.sortspec.md` could be implemented, if needed

**Comment by J3ster1337** [comment]
@SebastianMC Thank you a lot for such big answer! Yes, your code helped me with my problem. All I can say is this: <3

**Comment by SebastianMC** [comment, marked as answer]
This topic can be useful, re-opening the conversation to keep it visible to others
